crypto/src/memo.rs

```diff
@@ -140,23 +129,20 @@ mod tests {
     #[test]
     fn test_memo_encryption_and_decryption() {
```

Comment:
This is a good test! I think it should be pretty easy to convert it to a `proptest` that checks the condition:

> For any string, if its length in bytes is less than or equal to `MEMO_LEN_BYTES`, `string == decrypt(encrypt(string))` and if its length in bytes is greater than `MEMO_LEN_BYTES`, `encrypt(string)` returns an error.

That generalizes this test and the below, and might ferret out any corner cases we could be missing.

Reply:
If you haven't used `proptest` before, check out [The Proptest Book](https://altsysrq.github.io/proptest-book/intro.html).

Reply:
I feel relatively confident that if these two tests you already have pass, then things probably work, but there could be corner cases still, and I'm mostly commenting to point out the underappreciated fact that writing proptests is often little harder than writing single pointwise unit tests, and provides a huge amount of more confidence in correctness. So, when it's easy (and sometimes when it's hard) I use proptest all the time.

Reply:
So this is more of a nice opportunity for me to plug my favorite software assurance power tool, which I've used to great effect in a lot of places at Penumbra and elsewhere :)